tests/functional/lib/helpers.js

```diff
@@ -160,8 +236,24 @@ define([
    * @returns {promise} rejects if element does not exist
    */
   const testElementExists = thenify(function (selector) {
+    var findError;
     return this.parent
       .findByCssSelector(selector)
+      .then(null, function (err) {
+        // swallow the error until after taking a screenshot.
+        // The error has to be swallowed before a screenshot
+        // can be taken or else takeScreenshot itself returns
+        // the error.
```

Comment:
Purely an observation, but this comment surprises me after reading the code for `takeScreenshot`. It looks like `takeScreenshot` should resolve to `undefined`, is there some magic that I'm missing?